integration/insert_compat_test.go

```diff
@@ -68,30 +72,39 @@ func testInsertCompat(t *testing.T, testCases map[string]insertCompatTestCase) {
 						compatErr = UnsetRaw(t, compatErr)
 
 						// Skip inserts that could not be performed due to Tigris schema validation.
-						var e mongo.CommandError
-						if errors.As(targetErr, &e) && e.Name == "DocumentValidationFailure" {
-							if e.HasErrorCodeWithMessage(121, "json schema validation failed for field") {
-								setup.SkipForTigrisWithReason(t, targetErr.Error())
-							}
+						var e mongo.BulkWriteException
+						if errors.As(targetErr, &e) &&
+							e.HasErrorCodeWithMessage(121, "json schema validation failed for field") {
+							setup.SkipForTigrisWithReason(t, targetErr.Error())
 						}
 
-						assert.Equal(t, compatErr, targetErr)
+						assert.Equal(t, compatErr.Error(), targetErr.Error())
```

Comment:
Why that change? Why do we compare strings instead of error values?

Reply:
LGTM overall, but that change looks sus

Reply:
@AlekSi that was a bad idea indeed! Fixing... (not only this thing but also the problems it hided)

Reply:
fixed